src/Components/Web/src/Forms/EditContextModelBindingContextExtensions.cs

```diff
@@ -0,0 +1,102 @@
+{
+    private static readonly object _key = new();
+
+    /// <summary>
+    /// Enables <see cref="ModelBindingContext"/> errors to be added to the <see cref="EditContext"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="EditContext"/>.</param>
+    /// <param name="bindingContext">The <see cref="ModelBindingContext"/>.</param>
+    /// <returns></returns>
+    public static IDisposable EnableBindingContextExtensions(this EditContext context, ModelBindingContext bindingContext)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(bindingContext, nameof(bindingContext));
+
+        context.Properties[_key] = bindingContext;
+
+        return new BindingContextEventSubscriptions(context, bindingContext);
+    }
+
+    /// <summary>
+    /// Gets the attempted value for the specified field name.
+    /// </summary>
+    /// <param name="context">The <see cref="EditContext"/>.</param>
+    /// <param name="fieldName">The field name.</param>
+    /// <returns></returns>
+    public static string? GetAttemptedValue(this EditContext context, string fieldName)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(fieldName, nameof(fieldName));
+
+        if (context.Properties.TryGetValue(_key, out var result) && result is ModelBindingContext bindingContext)
+        {
+            return bindingContext.GetAttemptedValue(fieldName);
+        }
+
+        return null;
+    }
+
+    private sealed class BindingContextEventSubscriptions : IDisposable
+    {
+        private readonly EditContext _editContext;
+        private readonly ModelBindingContext _bindingContext;
+        private ValidationMessageStore? _messages;
+
+        public BindingContextEventSubscriptions(EditContext editContext, ModelBindingContext serviceProvider)
+        {
+            _editContext = editContext;
+            _bindingContext = serviceProvider;
+
+            _editContext.OnValidationRequested += OnValidationRequested;
+        }
+
+        private void OnValidationRequested(object? sender, ValidationRequestedEventArgs e)
+        {
+            if (_messages != null)
+            {
+                // We already added the messages from the binding context,
+                // we don't have to do anything.
+                return;
+            }
+
+            _messages = new ValidationMessageStore(_editContext);
+            var adddedMessages = false;
+            foreach (var ((owner, key), errors) in _bindingContext.GetAllErrors())
+            {
+                FieldIdentifier fieldIdentifier;
+                fieldIdentifier = new FieldIdentifier(owner ?? _editContext.Model, key);
+
+                foreach (var error in errors)
+                {
+                    adddedMessages = true;
+                    // TODO: We need to support localizing the error message.
```

Comment:
Would that be done here, or upstream in the model binding system? Do you have a sense of what the API would look like?

Reply:
I was thinking that we would expose a parameter on the `ModelBindingValidator` component, but that's challenging now since it's internal and we wire-it up automatically.

What are your preferences.